packages/expo/src/environment/logging.fx.ts

```diff
@@ -24,10 +24,34 @@ if (__DEV__) {
         Constants.manifest.logUrl = logUrl;
       }
     }
+  } else if (
+    // If this is defined then we can be define Constants.manifest.logUrl without worrying about the warning.
+    Constants.__unsafeNoWarnManifest2 &&
+    // Only attempt to set the URL if `Constants.__unsafeNoWarnManifest.logUrl` is not defined.
+    !Constants.__unsafeNoWarnManifest2.extra?.expoGo?.logUrl
+  ) {
+    const devServerInfo = getDevServer();
+    // Ensure the URL is remote and not local. i.e `file://`
+    if (devServerInfo.bundleLoadedFromServer) {
+      // url: `http://localhost:8081/`
+      const url = !devServerInfo.url.endsWith('/') ? `${devServerInfo.url}/` : devServerInfo.url;
+      // The standard Expo logUrl is `http://localhost:19000/logs`, this code assumes that the `logs` endpoint doesn't change.
+
+      const logUrl = url + 'logs';
+      if (Constants.__unsafeNoWarnManifest2.extra?.expoGo) {
+        Constants.__unsafeNoWarnManifest2.extra.expoGo.logUrl = logUrl;
+      }
+      if (Constants.manifest2?.extra?.expoGo) {
+        Constants.manifest2.extra.expoGo.logUrl = logUrl;
+      }
```

Comment:
in practice it probably doesn't make a difference but should we define `extra` or `extra.expoGo` if they aren't already in the manifest?

Reply:
I'd say probably not, though we'll see if this causes issues. I'm hesitant to mutate the manifest in general, but figured that this code was here for a reason so we probably needed to do it.